Cabal/Distribution/Simple/Utils/Json.hs

```diff
@@ -0,0 +1,34 @@
+module Distribution.Simple.Utils.Json
+    ( Json(..)
+    , renderJson
+    ) where
+
+data Json = JsonArray [Json]
+          | JsonBool !Bool
+          | JsonNull
+          | JsonNumber !Int
+          | JsonObject [(String, Json)]
+          | JsonString !String
+
+renderJson :: Json -> ShowS
+renderJson (JsonArray objs)   =
+  surround "[" "]" $ intercalate "," $ map renderJson objs
+renderJson (JsonBool True)    = showString "true"
+renderJson (JsonBool False)   = showString "false"
+renderJson  JsonNull          = showString "null"
+renderJson (JsonNumber n)     = shows n
+renderJson (JsonObject attrs) =
+  surround "{" "}" $ intercalate "," $ map render attrs
+  where
+    render (k,v) = (surround "\"" "\"" $ showString k) . showString ":" . renderJson v
+renderJson (JsonString s)     = surround "\"" "\"" $ showString s
+
+surround :: String -> String -> ShowS -> ShowS
+surround begin end middle = showString begin . middle . showString end
+
+intercalate :: String -> [ShowS] -> ShowS
+intercalate sep = go
+  where
+    go []     = id
+    go [x]    = x
+    go (x:xs) = x . showString sep . go xs
```

Comment:
@kanishka-azimi This part of the patch is actually going to vanish anyway, since we already have a aeson-ish mini library in `cabal-install` (which generates `plan.json`).  But more generally, there is IMO no point to release a small trivial mini-aeson library unless somebody other than cabal wants to actually use it. But I don't see this happening, as `aeson` is so ubiquitous that virtually everyone just uses `aeson` if they need JSON capabilities. Last but not least, there's a significant cost involved in maintaining a library on Hackage in a responsible manner (writing proper documentation, testing, keeping it working with updated dependencies and compilers, being careful about API breakges, reacting to feature requests and bug reports etc.).